task-sdk/src/airflow/sdk/execution_time/xcom.py

```diff
@@ -40,6 +40,7 @@ def set(
         task_id: str,
         run_id: str,
         map_index: int = -1,
+        mapped_length: int | None = None,
```

Comment:
Hmm, I don't love this being here - check out the comment on _xcom_push

Reply:
Valid call, i checked that comment out. The other option I can think of is calling the API directly for the case when mapped length is set? And not including it here? WDYT about that?

Reply:
I think I'll have to get over it - cos even when mapped length is set we need to store the value in the xcom backend. 

My brain isn't firing on all cylinders today, so do what you think is best.

Perhaps not document it and prefix it with `_`?

Reply:
Works as well, let me do that.

Reply:
Will revisit when needed